### PR TITLE
Multicluster link Helm chart enhancements

### PR DESCRIPTION
**Pull request by utay**
Hi there,

I'm adding 3 new fields to the linkerd-multicluster-link chart:
- `resources`
- `nodeSelector`
- `tolerations`

And also documentation in README for `targetClusterName`. 

Signed-off-by: Yannick Utard <yannick.utard@sorare.com>

### CODE REVIEW

File: multicluster/charts/linkerd-multicluster-link/values.yaml [hunk 1]
```diff
@@ -11,14 +11,22 @@ gateway:
...
+nodeSelector: {}
+# -- Resources for the Service mirror container
+resources: {}
 # -- Number of times update from the remote cluster is allowed to be requeued
 # (retried)
 serviceMirrorRetryLimit: 3
 # -- User id under which the Service Mirror shall be ran
 serviceMirrorUID: 2103
+# -- Name of the target cluster that's going to be linked
+targetClusterName: ~
```
Comment (adleong) [inline multicluster/charts/linkerd-multicluster-link/values.yaml:24]: I think an empty string is a better default/unset value here than "~".  The use of "~" suggests that this symbol has some special meaning or use.

**Review by adleong — COMMENTED** [review]
I will note that the link template is typically rendered by the `linkerd multicluster link` CLI command which does not set any of these newly added values.  This means that these values are only of use to people who are deploying this chart directly.

File: multicluster/charts/linkerd-multicluster-link/values.yaml (hunk 1 above)
Comment (utay) [reply multicluster/charts/linkerd-multicluster-link/values.yaml]: Makes sense, fixed!

**Review by adleong — APPROVED** [review]

**Comment by kleimkuhler** [comment]
Let me know if I'm testing this incorrectly but it seems that these values are being formatted incorrectly.

To reproduce, I have the branch checked out and run `bin/helm-build`. I template an example with `nodeSelector` set

```shell
$ helm template ./multicluster/charts/linkerd-multicluster-link/ --set nodeSelector="kubernetes.io/os: linux"
```

In the resulting manifest, `kubernetes.io/os: linux` is wrapped with `'` when it should not be

```yaml
...
      serviceAccountName: linkerd-service-mirror-
      nodeSelector: 
        'kubernetes.io/os: linux'
```

If you try to install this, you'll see it's rejected because it's being treated as a string rather than a map

```shell
$ helm install linkerd-multicluster-link ./multicluster/charts/linkerd-multicluster-link/ --set nodeSelector='foo: bar' --set targetClusterName=test
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(Deployment.spec.template.spec.nodeSelector): invalid type for io.k8s.api.core.v1.PodSpec.nodeSelector: got "string", expected "map"
```

**Comment by kleimkuhler** [comment]
Nevermind I got this to work by changing the flag to `--set nodeSelector."kubernetes\.io/os=linux"`.

**Review by kleimkuhler — APPROVED** [review]
Thanks @utay!

**Pull request merged**